src/test/scala/mesosphere/marathon/core/task/update/impl/steps/ScaleAppUpdateStepImplTest.scala

```diff
@@ -13,6 +13,35 @@ import mesosphere.marathon.state.{ PathId, Timestamp }
 
 class ScaleAppUpdateStepImplTest extends AkkaUnitTest {
 
+  // used pattern matching because of compiler checks, when additional case objects are added to Condition
+  def scalingWorthy: Condition => Boolean = {
+    case Condition.Reserved | Condition.Created | Condition.Killing | Condition.Running |
+      Condition.Staging | Condition.Starting | Condition.Unreachable => false
+    case Condition.Error | Condition.Failed | Condition.Finished | Condition.Killed |
+      Condition.UnreachableInactive | Condition.Gone | Condition.Dropped | Condition.Unknown => true
+  }
+
+  val allConditions = Seq(
+    Condition.Reserved,
+    Condition.Created,
+    Condition.Error,
+    Condition.Failed,
+    Condition.Finished,
+    Condition.Killed,
+    Condition.Killing,
+    Condition.Running,
+    Condition.Staging,
+    Condition.Starting,
+    Condition.Unreachable,
+    Condition.UnreachableInactive,
+    Condition.Gone,
+    Condition.Dropped,
+    Condition.Unknown
+  )
+
+  val scalingWorthyConditions = allConditions.filter(scalingWorthy)
+  val notScalingWorthyConditions = allConditions.filterNot(scalingWorthy)
```

Comment:
I'm fine with this change but would prefer scala test tables to avoid duplicate implementations:

```scala
class ScaleAppUpdateStepImplTest extends AkkaUnitTest with TableDrivenPropertyChecks {


"ScaleAppUpdateStep" when {
  // format: OFF  
  val conditions = Table (
    ("condition",             "initially", "with previous failed condition", "with previous running condition"),
    (Condition.Failed),    true       , false,                                              true                                                ),
    (Condition.Running), false     , false,                                              false                                              )
    ...
    )
  }
  //format: ON

  forAll (conditions) { (condition, scalingWorthy, scalingWorthyAfterFailed, scalingWorthyAfterRunnig) =>
    s"receiving status update $condition" should {
      s"send no rescaling request when " in { ... } 
    }
  }
}
}
```

Something like that. Here is an example https://github.com/mesosphere/marathon/blob/master/src/test/scala/mesosphere/marathon/core/task/TaskStatusComparisonTest.scala#L16.

Github does not align so well.